overlord/snapstate/handlers_components.go

```diff
@@ -227,3 +240,169 @@ func (m *SnapManager) undoMountComponent(t *state.Task, _ *tomb.Tomb) error {
 
 	return m.backend.RemoveComponentDir(cpi)
 }
+
+func (m *SnapManager) doLinkComponent(t *state.Task, _ *tomb.Tomb) error {
+	// invariant: component is not in the state (unlink happens previously if necessary)
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	// snapSt is a copy of the current state
+	compSetup, snapsup, snapSt, err := compSetupAndState(t)
+	if err != nil {
+		return err
+	}
+	csi := compSetup.CompSideInfo
+
+	// Grab information for current snap revision
+	// TODO will this still be correct when jointly installing snap +
+	// components? Link for the snap should happen earlier. But we do not
+	// want to reboot until all components are linked to the initramfs has
+	// the information to moun them already.
```

Comment:
we need to discuss and think, this is not a blocker for this PR, as we need this kind of tasks when adding a components after the fact, but ideally when doing snap+components refresh or install that full state of the snap appears again with the components already in it